src/main/java/seedu/address/logic/parser/FindCommandParser.java

```diff
@@ -27,18 +32,39 @@ public FindCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_KEYWORD, PREFIX_FIELD);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_KEYWORD, PREFIX_FIELD)
+        // throws exception if no keywords are specified
+        if (!arePrefixesPresent(argMultimap, PREFIX_KEYWORD)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
         List<String> keywords = argMultimap.getAllValues(PREFIX_KEYWORD);
         Optional<String> field = argMultimap.getValue(PREFIX_FIELD);
+        String fieldString;
 
-        if (field.get().equalsIgnoreCase("name")) {
+        if (field.isEmpty()) {
+            fieldString = "";
+        } else {
+            fieldString = field.get().toLowerCase();
+        }
+
+        switch (fieldString) {
+        case "email":
+            return new FindCommand(new EmailContainsKeywordsPredicate(keywords));
+        case "course":
+            return new FindCommand(new CourseContainsKeywordsPredicate(keywords));
+        case "name":
             return new FindCommand(new NameContainsKeywordsPredicate(keywords));
+        case "person":
+        case "":
+            return new FindCommand(new PersonContainsKeywordsPredicate(keywords));
+        case "phone":
+            return new FindCommand(new PhoneContainsKeywordsPredicate(keywords));
+        case "studentid":
+            return new FindCommand(new StudentIdContainsKeywordsPredicate(keywords));
+        default:
+            return new FindCommand(new PersonContainsKeywordsPredicate(Collections.<String>emptyList()));
```

Comment:
mmhmm yea, I re-looked through the codcov's warnings when i created the PR, presume this might be due to none of the test cases "reaching" the default branch. I believe it's trivial as `new FindCommand(...)` call itself has been tested, but I'll look into adding 1 or 2 test cases to address the warning in a future commit!